### DISCUSSION TITLE
Hide Caps Lock Indicator when Caps Lock is Turned off

### DISCUSSION THREAD

**Discussion by tajhans** [monkeytypegame/monkeytype#2298] [Ideas] [answered]
I like how the there is a caps lock indicator however if stays even if you turn off caps lock. The only time it goes away is if you refresh the page or start the test. I have noticed that my friends get very confused when this happens and I think a fix for this would be great.

**Comment by Ferotiq** [comment, marked as answer]
That might be platform specific because it works for me on windows & chrome, on iPadOS & safari though it stays until I start typing (if I activated it before starting a test).

**Reply by tajhans** [reply]
Yeah that’s what I am seeing.

**Reply by Miodec** [reply]
Yeah, looks like WebKit sends a `keydown` when caps lock is enabled and `keyup` when disabled. Very weird. Will see if i can handle both cases.

**Reply by ghost** [reply]
On Fedora,
caps on => popup gets displayed immediately
caps off => popup stays till the next key press

The problem seems to be `getModifierState("CapsLock")` which returns true, even on keyup when pressing CapsLock.
Furthermore, when holding CapsLock the popup also won't be displayed until you press another key.

I tried using `KeyboardEvent.key` to toggle the popup.
That worked.
But you still have to use `getModifierState("CapsLock")` in order to track the state at all times.
I tried multiple variations but nothing worked the way it should.

There are 3 actions to check:
- popup gets displayed / removed immediately when pressing CapsLock
- popup stays while holding CapsLock
  - when turning off, popup has to vanish on release
- popup gets displayed when CapsLock got turned on outside the test area

The first two can be handled with the basic key checking (`event.key`).
The last one can only be tracked using `getModifierState("CapsLock")`.
But as soon as you integrate the state check, the whole logic gets messed up.

I am out of ideas.

**Comment by dev-shetty** [comment]
> I like how the there is a caps lock indicator however if stays even if you turn off caps lock. The only time it goes away is if you refresh the page or start the test. I have noticed that my friends get very confused when this happens and I think a fix for this would be great.

Yea I am facing the same issue, I am using Brave Browser on Garuda Linux.
It still exists today 08/Apr/2023.
